### DISCUSSION TITLE
How do you get a users name?

### DISCUSSION THREAD

**Discussion by f0rme** [mjs/imapclient#497] [Q&A] [answered]
How do I go about getting a users name?

For example, like the names listed in the GMAIL inbox image below. 

![image](https://user-images.githubusercontent.com/31146437/206368044-10c4532a-ce8e-4923-8452-db70ee8f42dd.png)

**Comment by mjs** [comment, marked as answer]
The sender name can be obtained by retrieving the `From` header of an email which will give you something like either `First Last <someone@email.com>` or just a plain address like `someone@email.com`. This can be parsed with email.utils.parseaddr to give you the separate name and address parts.

Alternatively if you have the full email body (say because you fetched it as `RFC822`), you can parse it with  with email.parser.HeaderParser which will give you an EmailMessage object. This has a headers attribute from which you can get the `From` header and the separate as above using `parseaddr`. 

https://docs.python.org/3/library/email.examples.html
